src/importlinter/api.py

```diff
@@ -37,3 +40,60 @@ def read_configuration(config_filename: str | None = None) -> dict:
         "session_options": user_options.session_options,
         "contracts_options": user_options.contracts_options,
     }
+
+
+class FailedToCreateReport(Exception):
+    """
+    Opaque exception raised by create_report if an internal exception is raised by create_report
+    (below), including if the exception occurs when reading the user's configuration.
+    """
+    pass
+
+
+def create_report(
+    config_filename: Optional[str] = None,
+    limit_to_contracts: tuple[str, ...] = (),
+    cache_dir: str | None | Type[NotSupplied] = NotSupplied,
+    is_debug_mode: bool = False,
+    # show_timings: bool = False,
+    # verbose: bool = False,
+) -> Report:
+    """
+    Create a report of the import contracts.
+
+    Args:
+        config_filename:    Filepath for the configuration (e.g. .importlinter, pyproject.toml).
+                            If not provided, the default location will be checked.
+        limit_to_contracts: If supplied, only report on contracts with the supplied ids.
+        cache_dir:          Filepath to cache directory. Pass None to explicitly disable caching; if
+                            omitted, import-linter's default cache directory will be used.
+        is_debug_mode:      Set to True to raise full internal exceptions with stack traces. By
+                            default, exceptions will be rescued and instead a simple opaque
+                            exception will be raised.
+
+    Raises:
+        FailedToCreateReport if an internal exception is raised.
+        Other exceptions may be raised if is_debug_mode is True.
+
+
+    Not yet implemented::
+        show_timings:       whether to show the times taken to build the graph and to check
+                            each contract.
+        verbose:            if True, noisily output progress as it goes along.
+    """
+    try:
```

Comment:
This duplicates some of the logic from `use_cases.lint_imports`. I think instead we should extract the relevant logic from `lint_imports` to a separate `produce_report` function in `usecases` which could be called directly from here, and from within `lint_imports` - what do you think?

Reply:
100% agree, will do!